### PR TITLE
Fixed heatpump water heater (HPWH) crash when Outdoor Air Node name is required but is blank

### PR DESCRIPTION
**Pull request by rraustad**
## Pull request overview

Blank inputs for air nodes can lead to crash in heat pump water heaters. For example, if the HPWH is outdoors and the outdoor and exhaust air node names are blank the simulation halts without errors. This fixes issue #5747 and #5765 . No diff's expected in example files.
### Work Checklist

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
- [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
- [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
  - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
### Review Checklist

This will not be exhaustively relevant to every PR.
- [x] Code style (parentheses padding, variable names)
- [x] Functional code review (it has to work!)
- [x] If defect, results of running current develop vs this branch should exhibit the fix
- [x] CI status: all green or justified
- [x] Performance: CI Linux results include performance check -- verify this
- [x] Unit Test(s)
- C++ checks:
  - [x] Argument types
  - [x] If any virtual classes, ensure virtual destructor included, other things
- IDD changes n/a

### CODE REVIEW

File: src/EnergyPlus/PackagedTerminalHeatPump.cc [hunk 1]
```diff
@@ -2971,14 +2971,16 @@ namespace PackagedTerminalHeatPump {
...
+		if ( ZoneEquipmentListNotChecked ) {
+			if( ZoneEquipInputsFilled ) {
+				ZoneEquipmentListNotChecked = false;
+				for ( Loop = 1; Loop <= NumPTUs; ++Loop ) {
+					if ( CheckZoneEquipmentList( PTUnit( Loop ).UnitType, PTUnit( Loop ).Name, CtrlZoneNum ) ) {
+						// save the ZoneEquipConfig index for this unit
+						PTUnit( Loop ).CtrlZoneNum = CtrlZoneNum;
+					} else {
+						ShowSevereError( "InitPTHP: Packaged Terminal Unit=[" + PTUnit( Loop ).UnitType + ',' + PTUnit( Loop ).Name + "] is not on any ZoneHVAC:EquipmentList.  It will not be simulated." );
+					}
```
Comment (rraustad) [inline src/EnergyPlus/PackagedTerminalHeatPump.cc:2983]: This one-time check was executing every time through Init. Changed logical to static and re-organized to be more efficient.

File: src/EnergyPlus/WaterThermalTanks.cc [hunk 2]
```diff
@@ -811,6 +811,46 @@ namespace WaterThermalTanks {
...
+		return ErrorsFound;
+	}
+
+	bool
+	GetWaterThermalTankInputData(
+		bool & ErrorsFound
+	)
+	{
+
+		// SUBROUTINE INFORMATION:
```
Comment (rraustad) [inline src/EnergyPlus/WaterThermalTanks.cc:838]: Added upper level GetInput manager to enable unit testing.

File: src/EnergyPlus/WaterThermalTanks.cc [hunk 3]
```diff
@@ -2024,10 +2186,6 @@ namespace WaterThermalTanks {
 
 				} // DO HPWaterHeaterNum = 1, NumHeatPumpWaterHeater
 
-				if ( ErrorsFound ) {
-					ShowFatalError( "Errors found in getting " + cCurrentModuleObject + " input. Preceding condition causes termination." );
-				}
-
```
Comment (rraustad) [inline src/EnergyPlus/WaterThermalTanks.cc:2030]: Moved all fatal errors up to manager. "Previous errors" should show which object has problems.

File: src/EnergyPlus/WaterThermalTanks.cc [hunk 4]
```diff
@@ -811,6 +811,46 @@ namespace WaterThermalTanks {
 	{
 
 		// SUBROUTINE INFORMATION:
+		//       AUTHOR         Richard Raustad, FSEC
+		//       DATE WRITTEN   July 2016
+		//       MODIFIED       na
```
Comment (mjwitte) [inline src/EnergyPlus/WaterThermalTanks.cc]: For new functions, omit unused comments like this (all the na's and an headings that have na under them), and keep the other comment minimal.

File: src/EnergyPlus/WaterThermalTanks.cc [hunk 5]
```diff
@@ -1923,8 +1994,99 @@ namespace WaterThermalTanks {
...
+						} else {
+							if( HPWH.InletAirConfiguration == AmbientTempOutsideAir ) {
+								HPWHCoilOutletNodeNum = HPWH.ExhaustAirNode;
+							} else {
+								HPWHCoilOutletNodeNum = HPWH.HeatPumpAirOutletNode;
+							}
+						}
+						if ( DXCoilAirOutletNodeNum != HPWHCoilOutletNodeNum ) {
+							ShowSevereError( cCurrentModuleObject + "=\"" + HPWH.Name + "\":" );
+							ShowContinueError( "Heat pump water heater coil outlet node name not does not match next connected component." );
```
Comment (mjwitte) [inline src/EnergyPlus/WaterThermalTanks.cc]: Extra "not" in this one "node name not does not match . . ."

File: tst/EnergyPlus/unit/WaterThermalTanks.unit.cc [hunk 6]
```diff
@@ -1064,3 +1064,171 @@ TEST_F( EnergyPlusFixture, WaterThermalTank_CalcTempIntegral )
...
+		"    0.0,                     !- Coefficient3 x**2",
+		"    0.0,                     !- Minimum Value of x",
+		"    1.0;                     !- Maximum Value of x",
+
+	} );
+
+	ASSERT_FALSE( process_idf( idf_objects ) );
+
+	bool ErrorsFound = false;
+//	HeatBalanceManager::GetZoneData( ErrorsFound );
```
Comment (mjwitte) [inline tst/EnergyPlus/unit/WaterThermalTanks.unit.cc]: Since there are a couple other changes to make, this line can be deleted.

**Comment by mjwitte** [comment]
Tested defect files and HospitalLowEnergy.  All looks good and runtime is back to where it should be.  Just a few minor cleanups then this can drop in.

**Pull request merged**